### PR TITLE
59 - page 2 map improvements

### LINKED ISSUE #56: Page 2: refine the layout
Currently there's some layout issues on the LAD maps page:

- [ ] Maps are too small

image

- [ ] Table height is odd when on the tables tab (big gap to pagination)

image

### LINKED ISSUE #57: Page 2: add reset button for map focus
Currently we don't have a reset button on the maps, but we should add one so that users can reset their view if they drag it out to weird and wonderful places.

This kind of thing has worked in the past (and should go into dfeshiny):

```
#' addMapResetButton
#'
#' Add a map reset button to a Leaflet map object
#'
#' @param leaf A Leaflet map object
#'
#' @examples
#' reset_map <- wqTools::buildMap() %>% addMapResetButton()
addMapResetButton <- function(leaf) {
  leaf %>%
    addEasyButton(
      easyButton(
        icon = "ion-refresh",
        title = "Reset View",
        onClick = JS(
          "function(btn, map){ map.setView(map._initialCenter, map._initialZoom); }"
        )
      )
    ) %>%
    htmlwidgets::onRender(
      JS(
        "
function(el, x){
  var map = this;
  map.whenReady(function(){
    map._initialCenter = map.getCenter();
    map._initialZoom = map.getZoom();
  });
}"
      )
    )
}
```

Then when making the map you can just do:

``` 
map <- leaflet(...) %>%
     some stuff.... %>%
     addMapResetButton() %>%
```

### LINKED ISSUE #58: Page 2: give nice error message when no rows found
Currently if you select something that isn't available for the given filters it gives a beautiful error message:

image

We should make it should a genuinely nice error message instead when there are no rows for:
- [x] Provider table
- [x] Maps
- [x] Tables in the tables tab

### LINKED ISSUE #59: Page 2: Make it possible to select an LAD from the map
When you hover over the maps, you can currently see a tooltip.

image

It would be nice if a user could click on an LAD, the map zooms into that LAD and the other tables / map all filter as if that LAD has been selected in the relevant dropdown.

### LINKED ISSUE #61: Page 2: clean up data to remove 0's
Currently there's a lot of 0's in the data for the LAD tab, we should remove those as they're appearing in the map in an annoying way:

image

### PR DESCRIPTION
**Pull request by cjrace**
## Pull request overview

Updated the maps / page 2 in a few different ways to make them a bit neater for users. Resolves #56, resolves #57, resolves #58, resolves #59, resolves #61.

## Pull request checklist

Please check if your PR fulfills the following:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been run locally and are passing (`shinytest2::test_app()`)
- [x] Code is styled according to tidyverse styling (checked locally with `styler::style_dir()` and `lintr::lint_dir()`)

## What is the current behaviour?

Currently:

- Clicking on maps doesn't do anything
- Some 0's are showing
- Maps are too small
- There's a big gap to the tables pagination on the tables tab
- There's no way to reset the map if you've dragged it around
- Maps / tables will just error if there's no rows for the selections

## What is the new behaviour?

- Selecting an LAD from a map, triggers an LAD selection
- No 0's show on maps

- Maps have a reset button 
image

- Maps look a little neater - are larger in the space, have stronger blues 
image7

- The gap for tables has been removed
image

- Custom error message added for when there are no rows of data available for the selections made
image

## Anything else

I also renamed `dfe_map()` to `dfe_lad_map()` as I realised the function I'd made was LAD specific so felt that made more sense and would prevent confusion if trying to reuse it for any other levels.

Tests are failing, though that's expected because the fixes for them are still in #70.

### CODE REVIEW

**Comment by axcooper** [comment]
The map being selectable is a huge improvement, and the zeros being filtered out, resizing and better colours really help too. I am not able to get the reset button to work though?

**Comment by axcooper** [comment]
Think it would be even better if could select a provider using the table too?

**Comment by cjrace** [comment]
> The map being selectable is a huge improvement, and the zeros being filtered out, resizing and better colours really help too. I am not able to get the reset button to work though?

The reset button resets the zoom / focus if you move it about, though currently doesn't reset any LAD selection - is that what you're expecting to happen? (I've not done it before, but think that's possible to add if so)

Can look at adding the provider selection from the table too, probably in place of the current dropdown and then just having a search on the table like we do on a couple of the other tabs, won't get to it today but will add next week

**Comment by axcooper** [comment]
I was expecting the reset button to undo any LAD selection, if that is possible, that would be a great addition.

File: R/helper_functions.R [hunk 1]
```diff
@@ -172,8 +173,51 @@
   x
 }
 
+# Add a map reset button to a Leaflet map object, using some additional JavaScript
+# @param leaf A Leaflet map object
+# @param selectizeInputId The ID of the input you want the reset button to clear
+#
+# Note that if you're working inside a module, you will need to wrap the inputId in the NS function
+addMapResetButton <- function(leaf, selectizeInputId = NULL) {
```
[the remark this answers is not in the document]
Comment (cjrace) [reply R/helper_functions.R]: Fixed

File: R/helper_functions.R [hunk 2]
```diff
@@ -172,8 +173,51 @@
...
+    map._initialZoom = map.getZoom();
+  });
+}"
+      )
+    )
+}
+
 # Create a map ================================================================
-dfe_map <- function(data, measure) {
+dfe_lad_map <- function(data, measure, inputId) {
```
[the remark this answers is not in the document]
Comment (cjrace) [reply R/helper_functions.R]: Fixed

**Comment by cjrace** [comment]
I've made the reset button clear the dropdown input as well, takes a fraction of a second longer than I'd like but works well. Should be ready for you to test again @axcooper.

I have created #82 as a separate issue for making the table selectable, as I think it's probably best to cover that as a separate piece of work. Thinking of the time we have left and the value gained from doing it, as currently a user can do everything I think they needs, you have the dropdown for selecting or searching for the provider. Also thinking actually, it might be a nice test case to do some A / B testing to get some user feedback on a selectable table versus a dropdown search, which could then build evidence for why putting the extra effort in to make it selectable is worthwhile.

**Review by axcooper — APPROVED** [review]
Happy that the reset button works to clear the selected LAD

**Pull request merged**